meta-mender-core/classes/mender-part-images.bbclass

```diff
@@ -261,6 +261,29 @@ IMAGE_TYPEDEP_biosimg_append = "${@bb.utils.contains('IMAGE_FSTYPES', 'sdimg', '
 IMAGE_TYPEDEP_gptimg_append = "${@bb.utils.contains('IMAGE_FSTYPES', 'sdimg', ' sdimg', '', d)} \
                                ${@bb.utils.contains('IMAGE_FSTYPES', 'uefiimg', ' uefiimg', '', d)} \
                                ${@bb.utils.contains('IMAGE_FSTYPES', 'biosimg', ' biosimg', '', d)}"
+# Make sure the Mender part image is available in the live installer
+IMAGE_TYPEDEP_hddimg_append = "${@bb.utils.contains('IMAGE_FSTYPES', 'sdimg', ' sdimg', '', d)} \
+                               ${@bb.utils.contains('IMAGE_FSTYPES', 'gptimg', ' gptimg', '', d)} \
+                               ${@bb.utils.contains('IMAGE_FSTYPES', 'uefiimg', ' uefiimg', '', d)} \
+                               ${@bb.utils.contains('IMAGE_FSTYPES', 'biosimg', ' biosimg', '', d)}"
+
+# Use the Mender part image as the Live image
+python() {
+    if bb.utils.contains('IMAGE_FSTYPES', 'sdimg', True, False, d):
+        type='sdimg'
+    elif bb.utils.contains('IMAGE_FSTYPES', 'uefiimg', True, False, d):
+        type='uefiimg'
+    elif bb.utils.contains('IMAGE_FSTYPES', 'biosimg', True, False, d):
+        type='biosimg'
```

Comment:
Jenkins revealed that you will need to add `gptimg` here too. Maybe it's a good idea to put this at the end though, in case the user isn't building an image, but only an artifact:

```
else:
    return
```

otherwise he may get a `UnboundLocalError` for the `type` variable, as in the currently failing build. I would perhaps try a local build and remove all the `mender-image-*` using `MENDER_FEATURES_DISABLE_append`, just to be sure.

Reply:
Confirmed and fixed.